core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
         return super(FeedbackThreadUserModel, cls).get_multi(instance_ids)
 
 
+class GeneralFeedbackThreadModel(base_models.BaseModel):
+    """Threads for each entity.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # The type of entity the thread is linked to.
+    entity_type = ndb.StringProperty(required=False, indexed=True)
+    # The ID of the entity the thread is linked to.
+    entity_id = ndb.StringProperty(required=False, indexed=True)
+    # ID of the user who started the thread. This may be None if the feedback
+    # was given anonymously by a learner.
+    original_author_id = ndb.StringProperty(indexed=True)
+    # Latest status of the thread.
+    status = ndb.StringProperty(
+        default=STATUS_CHOICES_OPEN,
+        choices=STATUS_CHOICES,
+        required=True,
+        indexed=True,
+    )
+    # Latest subject of the thread.
+    subject = ndb.StringProperty(indexed=False)
+    # Summary text of the thread.
+    summary = ndb.TextProperty(indexed=False)
```

Comment:
Presumably also required. Not sure whether empty field fails the check though. Can you check whether required is just to exclude None or whether it excludes empty strings as well? If the latter then ignore this comment.

Reply:
I checked what you said. With summary='' no error occurs and the object is written. With summary=None, it throws an exception though. So I left it as is.

Reply:
Wait, sorry. I don't understand. The scenario you're describing is intended behaviour. Is that when "required" was True? If so then required should be True.

If on the other hand that's what happens when required = False, then what changes when required=True?

Reply:
With required=True, at least an empty string needed to be passed in. 

Oh I had misunderstood your review comment, I'll add required=True.

Reply:
Hmmm... I tried just making it required and I ran into a lot of test failures on feedback_services_test. This would mean that this wasn't guarded against in the past (and I also didn't see this field being populated in create_thread). So I think we would need to audit the existing threads and check that they all have a summary and then we need to make this required. Else, our migration would fail.

That being said, I have no idea what this field is being used for. I did a grep for thread.summary (assuming the common variable name for a FeedbackThread is "thread") and I didn't get any matches. I'll add this also to #5371? and do the check later?

Reply:
Yep that's fine. Thakns!